packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -199,9 +205,11 @@ function createHeadInsertionTransformStream(
     },
     async flush(controller) {
       // Check before closing if there's anything remaining to insert.
-      const insertion = await insert()
-      if (insertion) {
-        controller.enqueue(encoder.encode(insertion))
+      if (hasBytes) {
+        const insertion = await insert()
+        if (insertion) {
+          controller.enqueue(encoder.encode(insertion))
+        }
       }
```

Comment:
NIT: Maybe cleaner to read if we early-exit?

```suggestion
      if (!hasBytes) return

      const insertion = await insert()
      if (!insertion) return
      
      controller.enqueue(encoder.encode(insertion))
```

Reply:
IMO it reads less clear this way. The existence of additional html here is unusual and we're only checking it for correctness / completeness